kiali/kiali
# Adding HTTPS and TLS into K8s Gateway creation wizard

### Describe the change
In K8s Gateway creation wizard, added support of 'HTTPS'  protocol and 'TLS Certificate' input.
Done in a scope of GRPC Routing https://gateway-api.sigs.k8s.io/guides/grpc-routing/

![Screenshot from 2024-05-23 14-47-13](https://github.com/kiali/kiali/assets/604313/42a0c8f2-c5e7-4c6b-a6f9-a626a0472b10)

### Steps to test the PR

Istio config list -> Actions -> Create K8sGateway.
Add listener.
Choose HTTPS.
New row appears for TLS params. Currently only 'Terminate' is supported.
Add TLS Certificate.
Preview and save.
In create config object should be 'tls:' field with Kind and name. The 'Terminate' default mode value is not shown, considered as default for 'HTTPS'.
Regression testing: Make sure that "HTTP" protocol mode creation is also working fine.

### Automation testing

cypress test added

### Issue reference
https://github.com/kiali/kiali/issues/7223
Parent epic https://github.com/kiali/kiali/issues/7355
